tests/test_module.py

```diff
@@ -1425,6 +1426,294 @@ def test_dump_trimmed_series(self):
+                        assert sliced_actual_list2 == [[60L, '34635.4899999999'], [70L, '46593.49'], [80L, '60351.49'], [90L, '75909.49'], [100L, '93267.49']]
+                    elif agg == 'var.s':
+                        assert sliced_actual_list1 == [[60L, '743.611111111111'], [70L, '743.611111111111'], [80L, '743.611111111111'], [90L, '743.611111111111'], [100L, '743.611111111111']]
+                        assert sliced_actual_list2 == [[60L, '38483.8777777777'], [70L, '51770.5444444445'], [80L, '67057.2111111112'], [90L, '84343.8777777778'], [100L, '103630.544444444']]
+
+                    r.execute_command('DEL', key)
+                    r.execute_command('DEL', agg_key)
+  
+    def test_del_sample(self):
+        with self.redis() as r:
+            quantity = 10001
+            del_q = 100
+
+            type_list = ['', 'UNCOMPRESSED']
+            for chunk_type in type_list:
+                r.execute_command('ts.create', 'del', chunk_type)
+                for i in range(0, quantity, 1):
+                    r.execute_command('ts.add del', i, i)
+
+                ooo_res = r.execute_command('ts.range del - +')
+                assert [10000L, '10000'] == r.execute_command('ts.get del')
+                assert quantity == len(ooo_res)
+                assert quantity == self._get_ts_info(r, 'del').total_samples
+
+                # delete multiple random samples
+                for i in range(del_q):
+                    r.execute_command('ts.del del', i * 100)
+                    assert quantity - i - 1 == self._get_ts_info(r, 'del').total_samples
+                ooo_res = r.execute_command('ts.range del - +')
+                assert quantity - del_q == len(ooo_res)
+                assert quantity - del_q == self._get_ts_info(r, 'del').total_samples
+                assert quantity - 1 == self._get_ts_info(r, 'del').last_time_stamp
+                assert [10000L, '10000'] == r.execute_command('ts.get del')
+
+                # delete last sample
+                r.execute_command('ts.del del', quantity - 1)
+                ooo_res = r.execute_command('ts.range del - +')
+                assert len(ooo_res) == 9900
+                assert quantity - del_q - 1 == self._get_ts_info(r, 'del').total_samples
+                assert quantity - 2 == self._get_ts_info(r, 'del').last_time_stamp
+                assert [9999L, '9999'] == r.execute_command('ts.get del')
+
+                # delete whole chunks
+                for i in range(3000, 7000):
+                    try:
+                        r.execute_command('ts.del del', i)
+                    except: 
+                        pass
+                res = r.execute_command('ts.range del - +')
+                assert len(res) == 5940
+
+                # delete all samples
+                for i in range(0, quantity, 1):
+                    try:
+                        r.execute_command('ts.del del', i)
```

Comment:
you can check after each delete the total_samples went down by 1

Reply:
We have it `assert quantity - i - 1 == self._get_ts_info(r, 'del').total_samples`

Reply:
done